src/m_binbuf.c

```diff
@@ -506,7 +506,7 @@ static int binbuf_expanddollsym(const char *s, char *buf, t_atom *dollar0,
     char c = *cs;
 
     *buf=0;
-    while (c && (c>='0' && c<='9' || c=='-'))
```

Comment:
the check whether 'c!=0' is to detect a terminating NULL-byte, rather than to prevent null-ptr access.

Reply:
Okay, but I'm a bit confused because if c = 0 then doesn't c>='0' already have that scenario covered?